Add proper fields ordering during schemas JSON marshalling

morzhanov opened the pull request:
## Issue
Golang scrambles the JSON output, it needs to be ordered correctly before returning.

## Solution
Ordered JSON serialized content output: 
- [x] updated all Marshal methods with ordered maps to preserve proper fields ordering in schemas
- [x] updated serializer to handle new methods

mihirpandya-greenops commented on workflowtrigger/util/auditlog/log.go:
```diff
@@ -113,23 +115,38 @@ func UnmarshallString(str string) Log {
+		mapObj.Set("uniqueVersionInstance", remLog.UniqueVersionInstance)
+		mapObj.Set("unhealthyResources", remLog.UnhealthyResources)
+		mapObj.Set("remediationStatus", remLog.RemediationStatus)
+		mapObj.Set("type", serializerutil.RemediationLogType)
```
For the remediation log and the deployment log, put the type as the first component.

morzhanov replied:
Moved `type` field on the first place in those objects.

mihirpandya-greenops commented on workflowtrigger/util/clientrequest/request_events.go:
```diff
@@ -1,7 +1,7 @@
 package clientrequest
```
For all the serialization in these methods, order does not matter. Is it possible to automatically insert all the keys using the OrderedMap (not add them explicitly)?

morzhanov replied:
Unfortunately, OrderedMap does not have a method for bulk insertion. If we do not need those objects to be ordered we can use a simple map here. Updated those methods with `map[string]interface{}` return value.

mihirpandya-greenops commented on workflowtrigger/util/event/kafka_events.go:
```diff
@@ -2,6 +2,7 @@ package event
 
```
For all the serialization in these methods, order does not matter. Is it possible to automatically insert all the keys using the OrderedMap (not add them explicitly)?

morzhanov replied:
Updated those methods with `map[string]interface{}` return value.

mihirpandya-greenops reviewed (approved):
Remove all the .DS_Store files from the PR

morzhanov commented:
> Remove all the .DS_Store files from the PR

Yeah, those are annoying mac os auto-generated files. Removed those files and added `.DS_Store` to the `gitignore`.

mihirpandya-greenops reviewed (approved):
Looks great!
